### PR TITLE
Add SQS and Elasticache support

### PR DESCRIPTION
**Pull request by parasitew**
Add SQS API and Elasticache API in billow. 

* Elasticache API allows search for available reserved cache node offerings and provisioned cache clusters.
* The optional parameters in the billow query are separated into a single session because SQS, EC2, DynamoDB and Elasticache use the same syntax for searching.

@wyao

### CODE REVIEW

File: src/main/main.iml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+<module type="JAVA_MODULE" version="4">
+  <component name="NewModuleRootManager" inherit-compiler-output="true">
+    <exclude-output />
+    <content url="file://$MODULE_DIR$">
+      <sourceFolder url="file://$MODULE_DIR$/java" isTestSource="false" />
+    </content>
+    <orderEntry type="inheritedJdk" />
+    <orderEntry type="sourceFolder" forTests="false" />
+  </component>
+</module>
```
Comment (wyao) [inline src/main/main.iml]: We don't want this.

File: README.md [hunk 2]
```diff
@@ -25,42 +25,21 @@ and start contributing!
...
-  - `id,publicIP,launchTime`
 
 ### /ec2/all ###
 Search all ec2 instances.
 
+### /ec2/sg ###
+Get ec2 security groups.
+
 ### /rds/all ###
 Search all RDS resources.
```
Comment (wyao) [inline README.md]: I think the use of the term "search" is confusing here. We aren't actually searching, we are listing. Could we just say list?

File: README.md [hunk 3]
```diff
@@ -25,42 +25,21 @@ and start contributing!
...
-
-  - `type`
-  - `az,type`
-  - `id,publicIP,launchTime`
 
 ### /ec2/all ###
 Search all ec2 instances.
 
+### /ec2/sg ###
+Get ec2 security groups.
```
Comment (wyao) [inline README.md]: What's an example of this? Why are we pulling sg out specifically?

We should keep the ec2 docs together. Organizationally this readme is hard to follow as it jumps around and not every end point supports queries. How we organize the readme layout should reflect that.

File: src/main/java/com/airbnb/billow/Handler.java [hunk 4]
```diff
@@ -243,17 +395,72 @@ private void handleComplexEC2(HttpServletResponse response,
...
+            final Object value = Ognl.getValue(compiled, queues);
+            if (value instanceof Boolean && (Boolean) value)
+                queues.add(queue);
+        }
+
+        return queues;
+    }
+
+    Collection<ElasticacheReservedCacheNodesOffering> listReservedCacheNodeOfferingFromQueryExpression(final String
+                                                                                                    expression, final AWSDatabase db)
```
Comment (wyao) [inline src/main/java/com/airbnb/billow/Handler.java]: This indentation seems kind of off.

File: README.md [hunk 5]
```diff
@@ -89,30 +139,74 @@ Attributes(``readCapacityUnits`` and ``writeCapacityUnits`` are pulled out of
...
+    cacheNodes: "[]",
+    autoMinorVersionUpgrade: true,
+    securityGroups: "[]",
+    replicationGroupId: null,
+    snapshotRetentionLimit: null,
+    snapshotWindow: null
+}
+```
+
+### /elasticache/reserved_cache_node_offering ###
```
Comment (wyao) [inline README.md]: I don't believe this is interesting to us as it does not describe our topology.

Comment (parasitew) [inline README.md]: Is there any other elasticache action that is helpful to us?

File: src/main/java/com/airbnb/billow/Handler.java (hunk 4 above)
Comment (parasitew) [inline src/main/java/com/airbnb/billow/Handler.java]: I am using the airbnb java style file but the variable name is too long, I will remove this cuz we don't need "cache node offerings" any more

File: README.md (hunk 3 above)
Comment (parasitew) [inline README.md]: I put these under "EC2 Query" section, it will be easier for us to add more ec2 or dynamo related queries in the future.

File: README.md (hunk 5 above)
Comment (wyao) [inline README.md]: I think the cluster information should be sufficient for now.

File: README.md [hunk 6]
```diff
@@ -25,50 +25,105 @@ and start contributing!
...
-Example: `daysOld`.
+### EC2 Query ###
 
-- `l`/`limit`: maximum number of records to return (defaults to none). Example: `10`.
+#### /ec2/sg ####
+List all ec2 security groups.
 
-- `f`/`fields`: comma-separated list of fields to display (defaults to all). Examples:
+#### /ec2/all ####
+List all ec2 instances.
```
Comment (wyao) [inline README.md]: It's a bit silly to have both `/ec2/all` and `/ec2` without optional params return the exact same result. That said, if we shore this up there is some chance that we break backward compatibility, so I'm okay to keep it.

**Comment by wyao** [comment]
LGTM
